File: TopCoder02/PHP/Main.php
<?php
    class InterestingParty{
    	public function bestInvitation($first, $second){
 
    		for($i=0;$i<count($first);$i++){
    			$dic[$first[$i]] = 0;
				$dic[$second[$i]] = 0;
    		}
			for($i=0;$i<count($first);$i++){
				$dic[$first[$i]] = $dic[$first[$i]] + 1;
				$dic[$second[$i]] = $dic[$second[$i]] + 1; 
			}
			$ans = 0;
			foreach ($dic as $value) {
				$ans = max($ans,$value);
			}
			return $ans;
    	} 
    }
	$first = array("fishing", "gardening", "swimming", "fishing");
	$second = array("hunting", "fishing", "fishing", "biting");
	$ip = new InterestingParty;
	echo $ip->bestInvitation($first, $second);
?>

File: TopCoder02/Ruby/Main.rb
class InterestingParty
  def bestInvitation(first,second)
    dic = {}
    for i in 0..first.size-1
      dic[first[i]] = 0;
      dic[second[i]] = 0;
    end
    for i in 0..first.size-1
      dic[first[i]] = dic[first[i]] + 1;
      dic[second[i]] = dic[second[i]] + 1;
    end
    ans = 0
    for value in dic.values
      ans = [ans, value].max
    end
    return ans
  end
end
first = ["fishing", "gardening", "swimming", "fishing"]
second = ["hunting", "fishing", "fishing", "biting"]
puts InterestingParty.new.bestInvitation(first,second)
